### ISSUE TITLE
[Bug] Unable to view Keypad on IOS on Input focus - Xamarin Forms - EmbeddedWebView

### ISSUE THREAD

**Issue by Coder-RKV** [AzureAD/microsoft-authentication-library-for-dotnet#2017]
**Which Version of MSAL are you using ?**
4.17.1

**Platform**
Xamarin Forms

**What authentication flow has the issue?**
Other?
On the IOS Simulators (haven't yet tested on the phone), unable to see the keypad opening for any input fields. Tried IOS 11 (13.6.1) and IOS 8 (13.6.1). I am using EmbeddedWebView to display.

**Is this a new or existing app?**
App is in development

**Expected behavior**
Keypad should open on the focus of the input field. It works as expected in Android.

**Actual behavior**
Keypad doesn't open up on focus of input field in IOS. In the bottom able to see a bar with Done link and up / down arrows to toggle between the input fields but the keypad itself is not showing up.

**Possible Solution**
Not sure

**Comment by jennyf19** [comment]
thanks @Coder-RKV can you provide screenshots or a recording of what you're seeing. thanks.

**Comment by Coder-RKV** [comment]
@jennyf19 Please find attached.

The input is in focus, but unable to see the keypad.
Just seeing a Done button to come out of the focus for the input.
![Simulator Screen Shot - iPhone 8 - 2020-08-27 at 21 02 39 2](https://user-images.githubusercontent.com/1075905/91510617-9da35480-e8ab-11ea-81a0-f33429a7de00.png)

**Comment by jennyf19** [comment]
@Coder-RKV I checked w/our iOS team, and this is an issue w/Apple. They reported the issue a few days ago and are waiting to hear back. Will mark as external for now. thanks for reporting.

**Comment by Coder-RKV** [comment]
@jennyf19 Thank you for confirmation. Just wanted to double check on one thing though. Will it work on publishing to the device?

Perhaps its a simulator only issue?

**Comment by jennyf19** [comment]
@Coder-RKV our iOS team first noticed on a device, and sent that feedback to Apple, so to answer your question, I doubt this will work on a device. If i get more info from them, i'll update this issue.

**Comment by Coder-RKV** [comment]
@jennyf19 Thanks. So in essence the library version is unstable / unusable for IOS as of now.

**Comment by jennyf19** [comment]
@Coder-RKV more like an issue w/Apple, not us. You can use system browser until things w/Apple get resolved.
@kaisong1990 is working w/Apple on this, & can chime in and answer more questions if needed.

**Comment by Coder-RKV** [comment]
@jennyf19 Thank you. BTW, observed the same for without EmbeddedWebView(true) i.e. the system browser (at least on the simulator).

**Comment by jennyf19** [comment]
@kaisong1990 - @Coder-RKV is reporting the same on simulator. Which os version are you using @Coder-RKV ?

@trwalke can you try this out (embedded & system browser) w/the same os version? thx.

**Comment by kaisong1990** [comment]
@Coder-RKV We are facing the very similar issue on iOS side (real device) when user tries to type in username/password in the webView, the keyboard shows as transparent. Discussed with the web team, and they confirmed there is no special handling before showing the keyboard. We have filled a feedback to Apple regarding this issue and pending their response at the moment.

**Comment by Coder-RKV** [comment]
@kaisong1990 Thank you for confirmation. Will keep an eye on this.

**Comment by Coder-RKV** [comment]
@kaisong1990 @jennyf19 Any updates? This is actually a critical blocker to use MSAL library for our project.

**Comment by jennyf19** [comment]
@Coder-RKV We are waiting to hear back from Apple. The issue is external to MSAL. @kaisong1990  will update when he has info.

**Comment by kaisong1990** [comment]
@Coder-RKV One more clarification from you. Based on the description, it said ```the keyboard is not showing```, and the screenshot shows the arrow and done button at the very bottom. Can you try to type cmd+k on simulator to see if you can see the keyboard?

**Comment by Coder-RKV** [comment]
@kaisong1990 oh my god... that's like a crazy moment... do you think it'll work on the device as well then?
I'll try to publish it and check it anyway

**Comment by kaisong1990** [comment]
@Coder-RKV This is a feature on iOS simulator to let developers see more content during development. It should not happen on real device

**Comment by Coder-RKV** [comment]
@kaisong1990 @jennyf19 On an iPhone X OS 13.5.1, I am able to view the number keypad and the email (alphanumeric) keypad. With the cmd x, I'm able to see in simulator too.

**Comment by jennyf19** [comment]
@Coder-RKV can we close this issue? @trwalke is going to investigate the other iOS/Apple issue and open a new issue if needed. sound good?

**Comment by Coder-RKV** [comment]
@jennyf19 Closing the issue as am able to view the input keypad.

Also, could you please provide a one liner for the open iOS/Apple Issue so that we can be aligned on the same and if it has any impact.

**Comment by jennyf19** [comment]
@kaisong1990  can provide a one-liner and @trwalke will update if MSAL .NET is impacted

**Comment by Coder-RKV** [comment]
@kaisong1990 @trwalke @jennyf19 One-liner and impact please

**Comment by kaisong1990** [comment]
The issue we saw earlier is that when user taps textfield from sign in flow on web view, for example, email address or password field, the keyboard window will present from bottom. However, the keyboard itself is transparent. We found this issue during one of our bug bash from real device. And there is no constant way to reproduce this issue. The issue can be fixed by reopen the app or navigate web page back and force. It happens very rare.

We have reported the issue to Apple, and didn't see any major impact so far.
